### PR TITLE
Fix extraction of symbols in lsp-ui-sideline

### PR DESCRIPTION
**Pull request by kljohann**
As `forward-symbol` places point just _past_ the current symbol and not at the beginning of the next symbol some symbols were missed and several work-around were in place (e.g. checking for "::" or subtracting 1 from the column (even though this is already done by `lsp--cur-column`).

In particular for the following code
```
bool Something::check() { return true; }
```
the previous implementation would place point at the following positions:
```
bool Something::check() { return true; }
^   ^         ^      ^          ^    ^
```
where `Something` would be ignored because of the `(not (looking-at-p "::"))` check.
The new implementation looks at
```
bool Something::check() { return true; }
^    ^          ^         ^      ^
```
and considers all pointed-to symbols.

This pull request also introduces skipping of symbols in comments and checking for the hover capability.

### CODE REVIEW

**Comment by kljohann** [comment]
@sebastiencs Sure, I just pushed a rebased version. Can you take a look?

File: lsp-ui-sideline.el [hunk 1]
```diff
@@ -386,45 +386,51 @@ to the language server."
...
+              (let* ((symbol (thing-at-point 'symbol t))
+                     (bounds (bounds-of-thing-at-point 'symbol))
+                     (parsing-state (syntax-ppss))
+                     (in-string (nth 3 parsing-state))
+                     (outside-comment (eq (nth 4 parsing-state) nil)))
+                ;; Skip strings and comments
+                (when (and symbol (not in-string) outside-comment)
+                  (push (list symbol line bounds (lsp--cur-position)) symbols))))
+            (dolist (entry symbols)
+              (cl-destructuring-bind (symbol line bounds position) entry
```
Comment (sebastiencs) [inline lsp-ui-sideline.el]: Since lsp-ui is using dash, can you use `-let` to destructure the variable ?

File: lsp-ui-sideline.el [hunk 2]
```diff
@@ -386,45 +386,51 @@ to the language server."
...
+                        (progn (forward-symbol -1)
+                               (>= (point) bol)))
+              (let* ((symbol (thing-at-point 'symbol t))
+                     (bounds (bounds-of-thing-at-point 'symbol))
+                     (parsing-state (syntax-ppss))
+                     (in-string (nth 3 parsing-state))
+                     (outside-comment (eq (nth 4 parsing-state) nil)))
+                ;; Skip strings and comments
+                (when (and symbol (not in-string) outside-comment)
+                  (push (list symbol line bounds (lsp--cur-position)) symbols))))
```
Comment (sebastiencs) [inline lsp-ui-sideline.el]: `lsp--cur-position` is using `line-number-at-pos` which can be very slow on big files.
Can you use the same logic that was done ? (Computing the current line before looping on every symbol and use it to send the position)

**Comment by kljohann** [comment]
Done @sebastiencs .

**Comment by sebastiencs** [comment]
Great. Thanks @kljohann !

**Pull request merged**